modules/javafx.controls/src/main/java/com/sun/javafx/scene/control/ControlAcceleratorSupport.java

```diff
@@ -311,4 +332,42 @@ private static ReadOnlyObjectProperty<? extends Control> getControlProperty(Obje
 
         return null;
     }
+
+    // We need to store all the listeners added to each ObservableList so that we can remove them. For this we need
+    // a map mapping ObservableList to various listeners such as ListChangeListeners, but the map needs to be a
+    // WeakHashMap.
+    // The ideal key to the WeakHashMap would be the ObservableList itself, except for the fact that its equals method
+    // compares the list contents. If a WeakIdentityHashMap existed, we could use that instead with the ObservableList
+    // as the key.
+    // We can't use an IdentityWrapper as the key to the HashMap because we need a strong reference (ideally from the
+    // ObservableList itself) to the IdentityWrapper else it'll be garbage collected.
+    // Since every ObservableList gets a ListChangeListener, we can use that as an IdentityWrapper and rely on the fact
+    // ObservableList has a strong reference to the ListChangeListener.
+    static class IdentityWrapperListChangeListener implements ListChangeListener<MenuItem> {
+
+        ObservableList<? extends MenuItem> innerList;
+
+        public IdentityWrapperListChangeListener(ObservableList<? extends MenuItem> list) {
+            this.innerList = list;
+        }
+
+        @Override
+        public void onChanged(Change<? extends MenuItem> c) {
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hashCode(innerList);
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (o == null || !(o instanceof IdentityWrapperListChangeListener)) return false;
```

Comment:
(o instanceof IdentityWrapperListChangeListener) should take care of the null case.
if(o instanceof IdentityWrapperListChangeListener that) {
  return innerList == that.innerList;
}
return false;

And also, could we have curly braces in all the if statements please?

Reply:
In general should we use new syntax like that instanceof syntax? JavaFX requires Java 17 now, but if changes were to be backported then using the new syntax makes it a bit more difficult.

Reply:
that will be "SEP" (someone else's problem) ;-)

fx requires jdk17 as a boot jdk per JDK-8276144

Reply:
Right. For newly added or modified code, it's fine to use JDK 17 language constructs and API, unless you know that the code will be backported right away. In particular, we have been using pattern-matching `instanceof` in other places.